lsr-role2collection.py

```diff
@@ -0,0 +1,443 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# (c) 2020 Matt Martz <matt@sivel.net>
+# GNU General Public License v3.0+
+#     (see https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Usage:
+# lsr-role2collection.py [--namespace NAMESPACE]
+#                        [--collection COLLECTION]
+#                        --src-path SRC_PATH
+#                        --dest-path DEST_PATH
+#                        --role ROLE
+#                        [-h]
+
+import argparse
+import os
+import re
+import fnmatch
+from shutil import copytree, copy2, ignore_patterns
+
+from pathlib import Path
+
+ROLE_DIRS = (
+    'defaults',
+    'files',
+    'handlers',
+    'meta',
+    'tasks',
+    'templates',
+    'vars',
+)
+
+PLUGINS = (
+    'action_plugins',
+    'become_plugins',
+    'cache_plugins',
+    'callback_plugins',
+    'cliconf_plugins',
+    'connection_plugins',
+    'doc_fragments',
+    'filter_plugins',
+    'httpapi_plugins',
+    'inventory_plugins',
+    'library',
+    'lookup_plugins',
+    'module_utils',
+    'netconf_plugins',
+    'shell_plugins',
+    'strategy_plugins',
+    'terminal_plugins',
+    'test_plugins',
+    'vars_plugins'
+)
+
+TESTS = (
+    'tests',
+    'molecule',
+    '.travis',
+)
+
+DOCS = (
+    'docs',
+    'design_docs',
```

Comment:
There is also the `examples` directory e.g. used by network - seems more documentation than anything else

Reply:
The examples dir was treated as extra and placed in the top fedora/system_roles dir. Now, the files in the examples of each role are in the docs/ROLE.
```
$ ls fedora/system_roles/docs/
certificate/  logging/	metrics/  nbde_client/	nbde_server/  network/	timesync/
```
I noticed there is another non-standard directory treated as extra which is  "scripts" from network. It contains just one python script `print_all_options.py`. And the script appears only here in the network code...
`./.travis/config.sh:    PYTHON2_EXCLUDES="tests/ensure_provider_tests.py,scripts/print_all_options.py"`
How shall we handle it? (So far, this is the only an example that falls into the case, but in general, what'd be the best way for the role?)

Reply:
hmm - not sure how to handle the network/scripts/print_all_options.py - looks like it is used to generate the docs?  @tyll

Reply:
Currently it is not integrated into anything but I only used it manually to create an overview for some items in the README.md.

Reply:
Thanks for your input, @tyll! So, the script is needed in the original roles repository, but not in the generated collections. I'm going to skip copying it.